model_benchmark/common.py

```diff
@@ -572,12 +572,72 @@ def cast_to_fp32(model, inputs):
     return model, inputs
 
 
+class DummyGradScaler:
+    def scale(self, loss):
+        return loss
+
+
 class BenchmarkRunner:
+    def __init__(self):
+        self.use_amp = False
+        self.grad_scaler = DummyGradScaler()
+        self.autocast = NullContext
+        self._args = None
+
+    def setup_amp(self):
+        if self.args.amp and self.args.training:
+            assert self.args.devices == ["cuda"], "AMP is supported only for CUDA"
+            self.grad_scaler = torch.cuda.amp.GradScaler()
+            self.autocast = torch.cuda.amp.autocast
```

Comment:
Nit: you can use generic autocast manager with `device="cuda"` https://pytorch.org/docs/stable/amp.html#torch.autocast